fix(uploads): passes id and data to read access func when accessing upload URLs

JessChowdhury opened the pull request:
### What?
When accessing an upload directly from the generated URL, the `read` access runs but returns undefined `id` and `data`. As a result, any access conditions that rely on `id` or `data` will fail and users cannot accurately determine whether or not to grant access.

### Why?
Accessing the file URL runs `packages/payload/src/uploads/endpoints/getFile.ts`. 
In this endpoint, we use `checkFileAccess()` from `packages/payload/src/uploads/checkFileAccess.ts`.
Within the `checkFileAccess` function we are only passing the `req` to `executeAccess()`.

### How?
Passes `id` and `data` to the `executeAccess()` function from `uploads/checkFileAccess`.

Fixes #11263

r1tsuu reviewed (dismissed):
Looks good, if we can pass these arguments I don't see why we shouldn't.

Having that said, I think they should instead return a `where` query from their `access` function, because it works not only for `/media/1` and `/media/file/filename.png`, but also for `/media` (in which you surely cannot access `data`), for example:
```ts
return { tenant: { equals: user.tenant } }
```

JarrodMFlesch commented:
I am not sure about this. I am not sure that data should even be included in read access args. Maybe we pass:
```ts
data: { 
  filename
}
```

This way users can look the file up if they really need to. We should note that in findByID we do not pass data, just the id of the document.

The issue I have with this, is the fact that we are querying the doc 2 times even if the user does not need the data for the doc in the access function. Once before access control and another time with the results of the access control function.

JarrodMFlesch reviewed (changes requested):
I am not sure about this. I am not sure that data should even be included in read access args. Maybe we pass:
```ts
data: { 
  filename
}
```

This way users can look the file up if they really need to. We should note that in findByID we do not pass data, just the id of the document.

The issue I have with this, is the fact that we are querying the doc 2 times even if the user does not need the data for the doc in the access function. Once before access control and another time with the results of the access control function.

JessChowdhury commented:
Thanks @JarrodMFlesch and @r1tsuu - ah good points, I have updated it to only pass `data: { filename }` and users can then make the request to get the full data from there. Removed the added `findByID` request also.
